reorder pages in sidebar

namiroues opened the pull request:
Pages on sidebar are in this order now:

Home
Getting Started
Learning MDAnalysis
Documentation
MDAKits and MDA-based tools
Contribute
Community
Events
Citations
About MDAnalysis
MDAnalysis team
Blog
Privacy Policy

namiroues committed:
reorder pages in sidebar

orbeckst reviewed (approved):
Fine with me.

(Eventually it would be good to condense even more or use foldable submenus.)

namiroues committed:
reorder pages in sb- rename blog to news - move core pages to top level

micaela-matta reviewed (approved):
Thanks @namiroues for implementing the changes we discussed in our meeting 🙏

namiroues committed:
fix capitalisation consistency and rename Blog to News pages

namiroues commented:
Following up on the meeting suggestions by @micaela-matta and @jennaswa the new order is now:

Home
About MDAnalysis
News (renamed from Blog for clarity - waiting feedback from @orbeckst on this before changing all files)
Getting Started 
Learning MDAnalysis
Documentation
Events
Community
Contribute
MDAnalysis Team
MDAKits (shortened name for clarity - other possible options "MDAKits and extensions", or "MDAKits and dependencies") 
Citations
Privacy Policy

Also, the core pages have been moved to the top level.

orbeckst reviewed (approved):
Changes look fine to me. I like the lifting of core pages to top level. It just has the potential to break a lot of links. Some of these will be external links into our site and we'll just have to deal with this as we go along but if there's a way to check the whole site for broken links after the change then we should do that – can you raise an issue for doing that, please?

Could you please also check the wiki pages that deal with GSOC and update links there, in case they break?

micaela-matta commented on _posts/2023-05-18-save-date-ugm2023.md:
```diff
@@ -20,10 +20,10 @@ Whether you are a new MDAnalysis contributor, an MDAnalysis core developer, or a
 
 [uniL]: https://www.ulisboa.pt/en/unidade-organica/faculty-sciences
-[blog]: https://www.mdanalysis.org/blog/
+[News]: https://www.mdanalysis.org/blog/
```
it was the best we could come up with to convey the content of the blog :)

jennaswa commented:
Thanks, @namiroues ! Looking at it all together (and sorry to get so nitpicky on these things), Events seems like a sub-category to Community to me, and would make slightly more sense to me just below Community, but I don't feel too strongly here. If we do ultimately get around to doing a collapsable sidebar, this could be an example of where pages could potentially be condensed/combined further.

namiroues committed:
swap order between community and events pages

namiroues commented:
I resolved the conflicts @micaela-matta. Could you please check and merge? Thank you.

jennaswa reviewed (changes requested):
Thank you so much for the hard work on this @namiroues! I think it looks great and am happy with the main goal of this PR (i.e., the sidebar order). However, I've left a couple of suggestions/comments for resolution before I would say this is ready to merge.

namiroues commented on about.md:
```diff
@@ -14,8 +14,8 @@ The MDAnalysis Project is committed to open-source, reproducible computational r
-- **Join the Community:** Connect with other users, report issues, and contribute through multiple channels. Learn more on our [Community]({{ site.baseurl }}/pages/community/) page.
-- **Contribute:** Whether through coding, documentation, community support, or event organization, you can help shape the MDAnalysis ecosystem. Get started with our [Contributing Guide]({{ site.baseurl }}/pages/contribute/) or [join a team]({{ site.baseurl }}/pages/team/) to support specific initiatives. 
+- **Join the Community:** Connect with other users, report issues, and contribute through multiple channels. Learn more on our [Community]({{ site.baseurl }}/community/) page.
+- **Contribute:** Whether through coding, documentation, community support, or event organization, you can help shape the MDAnalysis ecosystem. Get started with our [Contributing Guide]({{ site.baseurl }}contribute/) or [join a team]({{ site.baseurl }}team/) to support specific initiatives. 
```
Yes, links break without the leading slash. I am going through all links to update. Apologies for missing this.

namiroues commented on contribute.md:
```diff
@@ -74,7 +74,7 @@ The [API reference]({{ site.docs.mdanalysis.url }}/stable/index.html) is **autom
 MDAnalysis has actively participated in various mentoring and outreach programs **to help students, researchers, and early-career contributors gain experience in open-source development and scientific software**. Through these programs, participants receive mentorship, contribute to real-world projects, and develop valuable skills in computational science and software engineering.
```
I think we can revisit this after merging the current branch and open a separate issue to go through the pages and check if everything fits properly. It feels a bit overwhelming to include all these adjustments as part of the sidebar reordering branch.

orbeckst replied:
Raising a new issue would be great — just do that and then I think we can count this issue as resolved for the purpose of this PR.

namiroues replied:
Issue #476 has been created.

namiroues commented on contribute.md:
```diff
@@ -110,7 +110,7 @@ MDAnalysis has participated in Google Summer of Code for several years, mentorin
 - [GSoC 2016]({% post_url 2016-04-26-gsoc %})
```
Could this be another issue?

jennaswa replied:
I think the latter part certainly could be (i.e., it could be looped with the issue mentioned in the comment above to generally update the mentoring section of the website). However, I've already linked to some of the project write-ups in the first half of this comment, so if it's easy enough to incorporate that change in this PR, I think why not :)

jennaswa replied:
Ahh, I see now that you already updated the links - thanks!!! :)

namiroues committed:
address PR review comments

orbeckst commented:
@namiroues do you think you'll be able to complete this PR? You've already put a lot of work into it and it would be great to have the updates go live.

namiroues commented on documentation.md:
```diff
@@ -1,7 +1,7 @@
+order: 5
 ---
 
 The MDAnalysis library (`import MDAnalysis`) is open-source code, available under the [GNU Lesser General Public License, version 2.1 or later (LGPLv2.1+)](https://www.gnu.org/licenses/lgpl-2.1.html). MDAnalysis provides **comprehensive documentation** to help users at all levels&mdash;whether you're just getting started, developing new tools, or contributing to the project. From **step-by-step t ...
```
@IAlibay ?

namiroues committed:
address PR review comment about licence

namiroues committed:
resolve conflicts for mdakits page

jennaswa reviewed (approved):
LGTM -- thank you so much for all of your hard work on this, @namiroues! It looks to me that the major items have been addressed. I say let's get this merged, and create new issues/smaller PRs if there are any outstanding issues.

jennaswa commented:
@namiroues - Hope you are doing well! I wanted to check-in on whether you might have some time to resolve the conflicts in this PR so we can get this merged. Thanks again for all your work on this, and let us know if you need anything from our end!

orbeckst committed:
Merge branch 'reorder-pages-in-sidebar' of https://github.com/namiroues/MDAnalysis.github.io into reorder-pages-in-sidebar

orbeckst commented:
@jennaswa are you happy to merge?

orbeckst reviewed (changes requested):
I built it locally and I am not sure yet that this is correct.

This is what it looks like locally (upload of screenshot fails)

Home
- About MDAnalysis
- News
- Getting started
- Learning MDAnalysis
- MDAKits and MDA-based tools
- UGM2025
- Documentation
- Community
- Events
- Contribute
- MDAnalysis Team
- MDAKits
- Citations
- Privacy Policy

Donate

orbeckst committed:
updated MDAKits page

- removed superfluous page
- update the top-level one

orbeckst committed:
renamed MDAKits to MDA Ecosystem

orbeckst committed:
moved UGM2025 to Events

- removed UGM 2025 from sidebar
- add stub entry to events
- removed from list of upcoming events

orbeckst reviewed (approved):
I did some clean-up and I think it's ready from my side
- order is now as discussed in https://github.com/MDAnalysis/MDAnalysis.github.io/pull/458#issuecomment-2780606653
- renamed "MDAKits and MDA-based tools" to "MDA Ecosystem"
- moved UGM2025 to events

orbeckst commented:
Ill trye *jekyll-redirect-pages* as it's one of the supported plugins, see https://pages.github.com/versions.json

orbeckst committed:
add redirects for all moved pages

- use plugin https://github.com/jekyll/jekyll-redirect-from
- added redirects to all pages that were moved from pages/ to root

orbeckst committed:
update action versions
